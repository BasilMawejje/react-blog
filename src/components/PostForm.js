import React from 'react'
import { Field, reduxForm } from 'redux-form'

class PostForm extends React.Component {
    renderError = ({ error, touched }) => {
        if (error && touched) {
            return (
                <div className='ui error message'>
                    <div className='header'>{error}</div>
                </div>
            );
        }
    }

    renderInput = ({ input, label, meta }) => {
        return (
                <div className='field'>
                    <label>{label}</label>
                    <input {...input} autoComplete='off' />
                    {this.renderError(meta)}
                </div>
            );
    }

    onSubmit = (formValues) => {
        this.props.onSubmit(formValues);
    }

    render() {
        return (
            <form onSubmit={this.props.handleSubmit(this.onSubmit)} className='ui form error'>
                <Field name='title' component={this.renderInput} label='Enter title'/>
                <Field name='body' component={this.renderInput} label='Enter description' />
                <button className='ui button primary'>Submit</button>
            </form>
        );
    }
}

const validate = (formValues) => {
    let errors = {};

    if(!formValues.title) {
        errors.title = 'Please enter a title';
    }

    if(!formValues.body) {
        errors.body = 'Please enter a description';
    }
    
    return errors;
}

export default reduxForm({
    form: 'postForm',
    validate
})(PostForm);