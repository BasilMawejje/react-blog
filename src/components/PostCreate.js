import React from 'react'
import { connect } from 'react-redux'
import { createPost } from '../actions/index'

import PostForm from './PostForm'

class PostCreate extends React.Component {
    onSubmit = (formValues) => {
        this.props.createPost(formValues);
    }

    render() {
        return (
            <div>
                <h1>Create Post</h1>
                <PostForm onSubmit={this.onSubmit} />
            </div>
        );
    }
}

export default connect(null, { createPost })(PostCreate);