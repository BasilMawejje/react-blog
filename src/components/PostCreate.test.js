import React from 'react';
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux'
import { shallowToJson } from 'enzyme-to-json';

import PostCreate from './PostCreate'
import PostForm from './PostForm'

configure({ adapter: new Adapter() });

const mockStore = configureStore();
const store = mockStore({});

describe('PostCreate', () => {
    let wrapper; 

    beforeEach(() => {
        wrapper = shallow(
            <Provider store={store}>
                <PostCreate>
                    <PostForm />
                </PostCreate>
            </Provider>
        ).dive();
    });

    it('renders properly', () => { 
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
})