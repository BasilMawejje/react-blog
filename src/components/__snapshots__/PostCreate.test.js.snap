// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCreate renders properly 1`] = `
<Connect(PostCreate)>
  <ReduxForm
    destroyOnUnmount={true}
    enableReinitialize={false}
    forceUnregisterOnUnmount={false}
    form="postForm"
    getFormState={[Function]}
    keepDirtyOnReinitialize={false}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    submitAsSideEffect={false}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
    validate={[Function]}
  />
</Connect(PostCreate)>
`;
